docs/TrafficSplitting.md

```diff
@@ -0,0 +1,120 @@
+ - **RevisionName**: RevisionName of a specific revision to which to send a portion of traffic.
+ - **Tag**: Tag is optionally used to expose a dedicated url for referencing this target exclusively.
+ - **Percent**: Percent specifies percent of the traffic to this Revision.
+
+## Operations
+
+Operations required to perform traffic splitting use cases, for example: blue/green deployment, etc can be summarized as below:
+ - Tag targets
+ - Split traffic
+
+### Tag Targets
+
+User can tag traffic targets (revisions) to
+ - Create custom URLs
+ - Create a human-friendly reference to refer to a traffic target
+ - Create static routes and change the underlying revisions as needed
+ - Remove the custom URLs OR remove the tags of targets
+ - Remove a target from traffic block altogether
+
+
+#### Tagging commands
+
+#### 1. `kn service tag-target [RevisionName1:Tag1] [RevisionName2:Tag2]`
+
+ - Tags given revision with specified tag thereby creating a custom URL
+ - Does not change the traffic percent of the target
+ - Multiple `RevisionName:Tag` combination can be specified
+ - Multiple tags for same revision can be specified
+   - `kn service tag-target echo-v1:stable echo-v1:current`
+ - If the target doesn't exist in deployed service, it creates a target entry with 0% traffic portion
+ - For updating tags of existing traffic targets, use `kn service untag-target` followed by `kn service tag-target`, because
+    - The deployed service might have multiple tags referring to a `RevisionName`
+    - The specified `RevisionName:Tag` combination on CLI might request to have different tags for same revision
+
+#### 2. `kn service untag-target [RevisionName:Tag] [RevisionName [--all]] [Tag]`
+
+ - Takes a single reference of a traffic target
+ - Does not change the traffic percent of the target
+ - Can be specified either via
+   -  `RevisionName`, removes the tag for target where `RevisionName` matches, errors if multiple tags for a revision exists
+   -  `Tag`, removes the specified tag from traffic targets
+   -  `RevisionName:Tag` removes `Tag` for `RevisionName`, useful for case where a single revision is tagged with multiple tags
+   - `--all` flag can be optionally specified along with `RevisionName` to remove all tags for given revision
+ - If a target has 0% traffic portion and it is untagged, target is removed from traffic block altogether
+
+
+## Split Traffic
+
+A user can
+ - Specify the traffic percent portion to route to a particular target
+ - Adjust the traffic percent of existing traffic targets
+ - Reference a traffic target either by `RevisionName` or `Tag`
+
+### Traffic splitting command
+
+#### `kn service set-traffic serviceName [revisionName1=Percent] [Tag2=Percent] [Tag3=*]`
+
+ - Takes the first argument as service name to operate on
+ - Subsequent arguments specifies the traffic targets
+ - Traffic target can be referenced either via `RevisionName` or `Tag`
```

Comment:
This does not guarantee a unique look-up of the block in question

Reply:
wdym ? Because a tag can be named like a revision ? I consider this case super seldom and we should verify this and throw an error, if then the tag and revision differs (as said (a) its very unlikely that a revision and tag will ever have the same name and (b) if this should be the case its very likely that it specifies the same block).

If you mean that when the user specifies a tag or revision which is not present, then of course again an error should be thrown.

So IMO it does guarantee a unique lookup (to 99.99999%) and simplifies the UI considerably.

Reply:
I mean, since the revisionName is not unique, `[revisionName1=Percent]` is indeterminate.

Reply:
* If the revisionName appears once, the operation is performed
* If the revisionName appears 0 times, an error is thrown
* If the revisionName appears more than once, an error is thrown and the user is hinted that she has to use a tag (multiple revisions without tags doesn't make any sense, or ?)

Reply:
(just updated my comment above which overlooked a case)